client/src/Controller/Admin/Client/SearchController.php

```diff
@@ -30,49 +32,68 @@ public function __construct(
      * @Security("is_granted('ROLE_ADMIN') or has_role('ROLE_AD')")
      * @Template("@App/Admin/Client/Search/search.html.twig")
      *
-     * @param Request $request
-     *
      * @return array|string
      */
-    public function searchAction(Request $request)
+    public function searchAction(Request $request, ParameterStoreService $parameterStore)
     {
+        $featureFlag = $parameterStore->getFeatureFlag(ParameterStoreService::FLAG_PAPER_REPORTS);
+
+        /** @var EntityDir\User $user */
+        $user = $this->getUser();
+
         $searchQuery = $request->query->get('search_clients');
         $form = $this->createForm(SearchClientType::class, null, ['method' => 'GET']);
 
-
         if (null === $searchQuery) {
-            return $this->buildViewParams($form);
+            return $this->buildClientViewParams($form);
         }
 
         $form->handleRequest($request);
         if ($form->isSubmitted() && $form->isValid()) {
             $filters = $form->getData() + $this->getDefaultFilters($request);
         }
 
-        $clients = $this->restClient->get('client/get-all?' . http_build_query($filters), 'Client[]');
+        if (('1' === $featureFlag && EntityDir\User::ROLE_SUPER_ADMIN === $user->getRoleName())
+            || EntityDir\User::ROLE_BEHAT_TEST === $user->getRoleName()) {
+            $cases = $this->restClient->get('case/search-all?'.http_build_query($filters), 'array');
```

Comment:
Following on from recent patterns could we extract this to a function in `ClientApi`?